src/components/explorer/ExplorerHeader.tsx

```diff
@@ -0,0 +1,332 @@
+import { buttonStyles } from 'components/common/button/buttonStyles';
+import { TokensOverlap } from 'components/common/tokensOverlap';
+import { useTokens } from 'hooks/useTokens';
+import { Strategy, useGetStrategList } from 'libs/queries';
+import {
+  PairTrade,
+  Trending,
+  useTrending,
+} from 'libs/queries/extApi/tradeCount';
+import { Link } from 'libs/routing';
+import { useEffect, useRef } from 'react';
+import { toPairSlug } from 'utils/pairSearch';
+
+const useTrendingPairs = (trending?: Trending) => {
+  const { tokensMap } = useTokens();
+  if (!trending) return { isLoading: true, data: [] };
+  const pairs: Record<string, PairTrade> = {};
+  for (const trade of trending?.pairCount ?? []) {
+    pairs[trade.pairAddresses] ||= trade;
+  }
+  const list = Object.values(pairs)
+    .filter((pair) => !!pair.pairTrades_24h)
+    .sort((a, b) => b.pairTrades - a.pairTrades)
+    .splice(0, 3);
+
+  // If there are less than 3, pick the remaining best
+  if (list.length < 3) {
+    const remaining = Object.values(pairs)
+      .filter((pair) => !!pair.pairTrades_24h)
+      .sort((a, b) => b.pairTrades - a.pairTrades)
+      .splice(0, 3 - list.length);
+    list.push(...remaining);
+  }
+
+  // Sort again in case we had to add more
+  const data = list
+    .sort((a, b) => b.pairTrades - a.pairTrades)
+    .map((pair) => ({
+      pairAddress: pair.pairAddresses,
+      base: tokensMap.get(pair.token0.toLowerCase())!,
+      quote: tokensMap.get(pair.token1.toLowerCase())!,
+      trades: pair.pairTrades,
+    }));
+  return { isLoading: false, data };
+};
+
+interface StrategyWithTradeCount extends Strategy {
+  trades: number;
+}
+const useTrendStrategies = (
+  trending?: Trending
+): { isLoading: boolean; data: StrategyWithTradeCount[] } => {
+  const trades = trending?.tradeCount ?? [];
+  const list = trades
+    .filter((t) => !!t.strategyTrades_24h)
+    .sort((a, b) => b.strategyTrades - a.strategyTrades)
+    .splice(0, 3);
+
+  // If there are less than 3, pick the remaining best
+  if (list.length < 3) {
+    const remaining = trades
+      .filter((t) => !!t.strategyTrades_24h)
+      .sort((a, b) => b.strategyTrades - a.strategyTrades)
+      .splice(0, 3 - list.length);
+    list.push(...remaining);
+  }
```

Comment:
This is in case there are not 3 strategies with trades for the last 24h. Then we take the first 1, 2 or 3 strategies **without** trades, we add them to the table and reorder incase the strategy without trades the last 24h have more trades than the one with trades the last 24h

Reply:
But the `list` and `remaining` lists are spliced from the same result:

```typescript
const list = Object.values(pairs)
    .filter((pair) => !!pair.pairTrades_24h)
    .sort((a, b) => b.pairTrades - a.pairTrades)
    .splice(0, 3);
```
```typescript
const remaining = Object.values(pairs)
    .filter((pair) => !!pair.pairTrades_24h)
    .sort((a, b) => b.pairTrades - a.pairTrades)
    .splice(0, 3 - list.length);
```

This part is common to both of them:
```typescript
Object.values(pairs)
    .filter((pair) => !!pair.pairTrades_24h)
    .sort((a, b) => b.pairTrades - a.pairTrades)
```